### PR TITLE
hadash: remember scoll positions of menus; "silent" option for service calls

### PR DESCRIPTION
**Pull request by flaparoo**
This PR implements 2 suggestions by @bobrippling for the HA-Dashboard (hadash):
- return to the same position within the menu (after a HA query/call)
- a "silent" option for service calls

### CODE REVIEW

File: apps/hadash/hadash.app.js [hunk 1]
```diff
@@ -65,10 +69,14 @@ function callService(title, domain, service, data, level) {
     },
   }).then(data => {
     //console.log(data);
-    E.showPrompt('Service called successfully', { title: title, buttons: {OK: true} }).then((v) => { E.showMenu(menus[level]); });
+    if (silent) {
+      scroller = E.showMenu(menus[level]).scroller;
+    } else {
+      E.showPrompt('Service called successfully', { title: title, buttons: {OK: true} }).then((v) => { scroller = E.showMenu(menus[level]).scroller; });
+    }
```
[the remark this answers is not in the document]
Comment (flaparoo) [reply apps/hadash/hadash.app.js]: Done and ...

File: apps/hadash/hadash.app.js [hunk 2]
```diff
@@ -220,8 +232,8 @@ addMenuEntries(0, settings.menu);
 
 // check required configuration
 if (! settings.HAbaseUrl || ! settings.HAtoken) {
-  E.showAlert('The app is not yet configured!', 'HA-Dash').then(() => E.showMenu(menus[0]));
+  E.showAlert('The app is not yet configured!', 'HA-Dash').then(() => scroller = E.showMenu(menus[0]).scroller);
 } else {
-  E.showMenu(menus[0]);
+  scroller = E.showMenu(menus[0]).scroller;
```
[the remark this answers is not in the document]
Comment (flaparoo) [reply apps/hadash/hadash.app.js]: ... done.

Way more elegant - thanks for the suggestions @bobrippling

**Comment by bobrippling** [comment]
Works an absolute treat, thanks! I'm thinking about adding an option to have scrollbars in an entry to hadash, to change brightness/light temperature for example. Is this something you'd be ok with?

**Comment by flaparoo** [comment]
> Works an absolute treat, thanks! I'm thinking about adding an option to have scrollbars in an entry to hadash, to change brightness/light temperature for example. Is this something you'd be ok with?

Of course!
I'm thinking it would warrant its own "type" in the JSON config (eg" `"type": "slider"`), but very similar to the "service" type. Please feel free to pick my brain if needed.

**Pull request merged**